lib/object-utils.js

```diff
@@ -230,8 +233,15 @@
      *      the input objects are not changed in any way.
      */
     function mergeFileCoverage(first, second) {
-        var ret = JSON.parse(JSON.stringify(first)),
-            i;
+        var ret, i;
+        if (hasObjectAssign) {
+            ret = Object.assign({}, first);
+            ret.s = Object.assign({}, ret.s);
+            ret.f = Object.assign({}, ret.f);
+            ret.b = Object.assign({}, ret.b);
```

Comment:
That is a shallow merge. Are you sure that the returned object isn’t modified anywhere?